stages/org.osbuild.dmverity

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+import os
+import subprocess
+import sys
+import tempfile
+import osbuild.api
+
+
+def open_dmverity(blocksize, partlabel, datapart, hashpart, mountpoint):
+    hash = subprocess.run(["/usr/sbin/veritysetup",
+                            "format", datapart, hashpart,
+                            "--data-block-size", f"{blocksize}"],
+                           check=True, capture_output=True)
+    for line in hash.stdout.decode().splitlines():
+      if line.startswith("Root hash:"):
+        hash = line.split(":", 1)[1].strip()
+        break
+
+    subprocess.run(["/usr/sbin/veritysetup",
+                    "open", datapart, partlabel, hashpart, hash],
+                   check=True)
+    subprocess.run(["mount", "-o", "ro", f"/dev/mapper/{partlabel}", mountpoint],
+                   check=True)
+    return hash
+
+
+def close_dmverity(partlabel, mountpoint):
+    subprocess.run(["umount", mountpoint], check=False)
+    subprocess.run(["/usr/sbin/veritysetup", "close", partlabel], check=False)
+
+
+def main(tree, paths, devices, options):
+    blocksize = options.get("blocksize", 512)
+
+    disk = os.path.join(paths["devices"], devices["disk"]["path"])
+    partition = f"{disk}p{options["partition"]}"
+    hash_partition = f"{disk}p{options["hash_partition"]}"
+
+    label = options["label"]
+    hashfile = os.path.join(tree, f"{label}_hash")
```

Comment:
Out of curiosity, what will happen to the hashfile in the tree, i.e. how is this used when generating an image?

Reply:
Yes, in case of `Secure Execution` we  append hashes to kernel `cmdline`, so during boot we can verify the image or stop the boot process. 
https://github.com/coreos/coreos-assembler/pull/3764/files#diff-6887dfe7d9e3c9100cd3e726cab390b9f26521c4051bb24e1443f76ae68e8fa4R119